src/mlx/warnings.py

```diff
@@ -12,46 +13,167 @@
 junit_pattern = re.compile(JUNIT_WARNING_REGEX)
 
 
-class WarningsPlugin:
-    def __init__(self):
-        self.sphinx_counter = 0
-        self.doxygen_counter = 0
-        self.junit_counter = 0
+class WarningsChecker(object):
+
+    def __init__(self, name = None, pattern = None):
+        # type: (string, unicode) -> None
+        self.pattern = pattern
+        if name is None:
+            self.name = 'uninitialized'
```

Comment:
If you create a custom checker to use with WarningsPlugin and forgot to provide a name - then this is needed.